apps/blade/src/app/page.tsx

```diff
@@ -1,14 +1,53 @@
+import Link from "next/link";
+
+import { auth } from "@forge/auth";
+import { eq } from "@forge/db";
+import { db } from "@forge/db/client";
+import { Member } from "@forge/db/schemas/knight-hacks";
+
 import { HydrateClient } from "~/trpc/server";
 import { AuthShowcase } from "./_components/auth-showcase";
 
-export default function HomePage() {
+export default async function HomePage() {
+  const session = await auth();
+
+  // checks if the user is signed in
+  // if so, it checks to see if the user is a member
+  // if not, they are displayed a sign-up link
+  // otherwise, if they are a member, they will be shown the dashboard link
+  if (session) {
+    const member = await db.query.Member.findFirst({
```

Comment:
Should move this into a public tRPC procedure so its reusable. We'll probably reuse this logic a lot.